core/adapters/eth_tx.go

```diff
@@ -260,13 +291,22 @@ func getTxDataUsingABIEncoding(e *EthTx, inputData models.JSON) ([]byte, error)
 				var arg [32]byte
 				copy(arg[:], b)
 				values[i] = arg
+			} else if argType == "address" {
+				if !common.IsHexAddress(jsonValues[i].String()) || len(b) != 20 {
+					return nil, errors.Wrapf(ErrInvalidABIEncoding, "invalid address %v", jsonValues[i].String())
+				}
+				values[i] = common.HexToAddress(jsonValues[i].String())
+			} else if argType == "bytes4" {
+				if len(b) != 4 {
+					return nil, errors.Wrapf(ErrInvalidABIEncoding, "can't convert %v to %v", jsonValues[i].Value(), argType)
+				}
+				var arg [4]byte
+				copy(arg[:], b)
+				values[i] = arg
```

Comment:
And if argType is something else?

Reply:
Switch statement might help to clean this up

Reply:
it can't be - thats what this check is for https://github.com/smartcontractkit/chainlink/pull/3720/files/29bfa02e3fc08549fcfa0239df41e7ae0cb9129e#diff-aa28966094395064f5f9020fd8e8930a70dacb161f93548bf517248aa9a56c9aR269